domain/agentbinary/state/state.go

```diff
@@ -97,8 +97,8 @@ INSERT INTO agent_binary_store (*) VALUES ($agentBinaryRecord.*)
 			// There must be an agent version for this version and arch already.
 			// We do not want to overwrite this value as it could result in a
 			// security risk.
-			return errors.New(
-				"agent binary already exists",
+			return errors.Errorf(
+				"agent binary of %q already exists", agentBinary.Version,
```

Comment:
This information is already added outside of the closure below. It would be a double up of the same thing.